### PR TITLE
BUG: Do no install any Eigen3 targets if ITK_USE_SYSTEM_EIGEN=ON

### PR DESCRIPTION
**Pull request by phcerdan**
Also added documentation about the logic having two Eigen3 targets,
and a minimal CMake example for usage in external modules.

Raised in: https://github.com/Microsoft/vcpkg/pull/5115#issuecomment-471109666

### CODE REVIEW

File: Modules/ThirdParty/Eigen3/CMakeLists.txt [hunk 1]
```diff
@@ -12,18 +12,51 @@ set(_Eigen3_min_version 3.3)
...
-if(ITK_USE_SYSTEM_EIGEN)
-  set(_Eigen3_SYSTEM_OR_INTERNAL "Eigen3")
-else()
-  set(_Eigen3_SYSTEM_OR_INTERNAL "ITKInternalEigen3")
-endif()
+# Docs:
+# find_package(ITK REQUIRED COMPONENTS
+# ITKEigen3
+# )
+# When requiring finding this module, the following cmake is invoked:
```
Comment (thewtex) [inline Modules/ThirdParty/Eigen3/CMakeLists.txt]: Not sure how to interpret this -- "When requiring finding this module" -> "When this module is a dependency"?

File: Modules/ThirdParty/Eigen3/CMakeLists.txt [hunk 2]
```diff
@@ -12,18 +12,51 @@ set(_Eigen3_min_version 3.3)
...
+# find_package(Eigen3 CONFIG) if ITK_USE_SYSTEM_EIGEN=ON
+# Or
+# find_package(ITKInternalEigen3 CONFIG) otherwise
+#
+# ITKInternalEigen3 does NOT allow the use of #include <Eigen/Core>, it requires #include <itkeigen/Eigen/Core>
+# This logic is handled automatically by the macro used internally `#include ITK_EIGEN(Core)`
+
+# However, this does not scales well for external ITK modules that want
```
Comment (thewtex) [inline Modules/ThirdParty/Eigen3/CMakeLists.txt]: scales -> scale

File: Modules/ThirdParty/Eigen3/CMakeLists.txt [hunk 3]
```diff
@@ -12,18 +12,51 @@ set(_Eigen3_min_version 3.3)
...
+# to link to third party libraries that are already using Eigen3.
```
Comment (thewtex) [inline Modules/ThirdParty/Eigen3/CMakeLists.txt]: that -> which

File: Modules/ThirdParty/Eigen3/CMakeLists.txt (hunk 1 above)
Comment (phcerdan) [reply Modules/ThirdParty/Eigen3/CMakeLists.txt]: Thanks a lot Matt.
I went for: When this module is required

Let me know if When this module is a dependency reads better.

**Pull request merged**